package/kedro_viz/integrations/kedro/lite_parser.py

```diff
@@ -0,0 +1,265 @@
+                module_name = node.module if node.module else ""
+                level = node.level
+
+                # Ignore relative imports like "from . import a"
+                if not module_name:
+                    continue
+
+                # Ignore relative imports within the package
+                # Examples:
+                # "from demo_project.pipelines.reporting import test",
+                # "from ..nodes import func_test"
+                if (self._package_name and self._package_name in module_name) or (
+                    # dev or standalone mode
+                    not self._package_name
+                    and project_file_paths
+                    and self._is_relative_import(module_name, project_file_paths)
+                ):
+                    continue
+
+                # absolute modules in the env
+                # Examples:
+                # from typing import Dict, Union
+                # from sklearn.linear_model import LinearRegression
+                if level == 0:
+                    self._populate_missing_dependencies(
+                        module_name, missing_dependencies
+                    )
+
+        return missing_dependencies
+
+    def create_mock_modules(self, unresolved_imports: Set[str]) -> Dict[str, MagicMock]:
+        """Creates mock modules for unresolved imports
+
+        Args:
+            unresolved_imports (Set[str]): A set of unresolved imports
+
+        Returns:
+            A dictionary of mocked modules for the unresolved imports
+        """
+        mocked_modules: Dict[str, MagicMock] = {}
+
+        for unresolved_import in unresolved_imports:
+            mocked_modules[unresolved_import] = MagicMock()
+
+        return mocked_modules
+
+    def parse(self, target_path: Path) -> Union[Dict[str, Set[str]], None]:
+        """Parses the file(s) in the target path and returns
+        any unresolved imports for all the dependency errors
+        as a dictionary of file(s) in the target path and a set of module names
+
+        Args:
+            target_path (Path): The path to parse file(s)
+
+        Returns:
+            A dictionary of file(s) in the target path and a set of module names
+        """
+
+        if not target_path.exists():
+            logger.warning("Path `%s` does not exist", str(target_path))
+            return None
+
+        unresolved_imports: Dict[str, Set[str]] = {}
+
+        if target_path.is_file():
+            missing_dependencies = self._get_unresolved_imports(target_path)
+            if len(missing_dependencies) > 0:
+                unresolved_imports[str(target_path)] = missing_dependencies
+            return unresolved_imports
+
+        # handling directories
+        _project_file_paths = set(target_path.rglob("*.py"))
+
+        for file_path in _project_file_paths:
+            missing_dependencies = self._get_unresolved_imports(
+                file_path, _project_file_paths
+            )
```

Comment:
This is traversing everything, including virtual environments

Reply:
The check is inside _get_unresolved_imports, will move it out. But I wonder why it is still parsing venv

Reply:
@astrojuanlu fixed this. Please review it when you have some time. Thank you